server/pom.xml

```diff
@@ -52,6 +52,12 @@
             <groupId>org.springframework.security</groupId>
             <artifactId>spring-security-cas</artifactId>
             <version>6.3.3</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.bouncycastle</groupId>
+                    <artifactId>bcprov-jdk15on</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
```

Comment:
Tää taitaa estää haavoittuvan depsun päätymisen classpathiin ainoastaan tämän yhden depsun osalta? (exclusion on määritetty dependencyn scopessa)

Jos mikään muu depsu sattuu riippumaan samasta haavoittuvasta depsusta, päätyykö se silloin classpathiin?

Reply:
Vähän mietin onko tästä exclusionista varsinaisesti mitään iloa. Jos softa toimii ilman excludattua kirjastoa, se lienee aika selvä merkki ettei haavoittuvaa koodia ole käytössä.

Toki estämällä haavoittuvan depsun lataaminen kokonaan saadaan haavoittuvan koodin käyttö virhetilanteeksi, jos jokin koodimuutos ottaisi haavoittuvia osia käyttöön, joten on siitä ehkä _jotain_ hyötyä.

Reply:
Joo, vain tämän yhden depsun osalta, ymmärtääkseni. Toisaalta jos jokin muu depsu toisi tuon mukaan classpathiin niin IDEA varoittaisi sitten siitäkin. Dependabot hassusti ei varoita tästä ollenkaan.

Reply:
Hmm, osaako dependabot edes reagoida jos haavoittuvuuden korjaavaa versiota ei vielä ole julkaistu? Vai eikö se haista transitiivisten riippuvuuksien haavoittuvuuksia? :thinking:

Reply:
Periaatteessa hieman lisäturvaa voisi saada [kieltämällä haavoittuvan depsun käyttö](https://maven.apache.org/enforcer/enforcer-rules/bannedDependencies.html) kokonaan. En tosin tiedä onko noin raskaalla lekalla huitominen tarpeen :thinking:

Reply:
Tuo on kyllä se oikea ratkaisu tässä. Siihen voi sitten lisätä kommentin miksi otettiin näin raju ase käyttöön. Hyvä löytö.